### ISSUE TITLE
Can find an app, but not download it

### ISSUE THREAD

**Issue by DamienCassou** [matlink/gplaycli#16]
```
$ gplaycli --config credentials.conf --folder ~/Documents/projects/playstore --search "capitaine train"
Title                         Creator              Size     Downloads   Last Update   AppID                                               Version  Rating  
Captain Train: train tickets  Capitaine Train SAS  5.9MB    100,000+    Oct 12, 2015  com.capitainetrain.android                          1704     4.47    
OUIGO                         SNCF                 1.1MB    100,000+    Mar 24, 2015  com.sncf.ouigo                                      6        3.38    
Voyages-SNCF                  Voyages-sncf.com     11.9MB   1,000,000+  Oct 20, 2015  com.vsct.vsc.mobile.horaireetresa.android           2508     4.17    
SNCF Transilien               SNCF                 6.4MB    1,000,000+  Apr 13, 2015  com.ocito.sncf                                      30       3.71    
SNCF                          SNCF                 7.0MB    500,000+    Oct 22, 2015  com.sncf.fusion                                     77       3.29    
Melbourne Metro Train Status  lahsrah apps         334.4KB  10,000+     Aug 27, 2014  com.lahsrah.metroStatus                             6        3.64    
Tatto Camera Prank            Tomara Penchuk       4.7MB    1,000+      Oct 24, 2015  com.shigetsogrartoapro.tattocameraprank             1        3.93    
Tattoo Camera Prank           Sofia Pikova         4.6MB    1,000+      Oct 24, 2015  com.xisaboakrevishyfehdevelopers.tattoocameraprank  2        3.11    
Electric Screen Prank         Maksimka             1.8MB    1,000+      Oct 24, 2015  com.prodavirdodru.electricscreenprank               1        3.00    
Tattoo Camera Joke            Snegana Yrkevich     4.3MB    500+        Oct 23, 2015  com.developersepsooreerev.tattoocamerajoke          1        4.33    

$ gplaycli --config credentials.conf --folder ~/Documents/projects/playstore --download com.sncf.fusion --progress
Error while downloading com.sncf.fusion : this package does not exist, try to search it via --search before

$ gplaycli --config credentials.conf --folder ~/Documents/projects/playstore --download com.capitainetrain.android --progress
Error while downloading com.capitainetrain.android : this package does not exist, try to search it via --search before
```

**Comment by DamienCassou** [comment]
I can still confirm the problem.

**Comment by raphael0202** [comment]
What are your credentials? I used the default ones on a freshly cloned repo, and both downloads work.

**Comment by DamienCassou** [comment]
Default credentials result in https://github.com/matlink/gplaycli/issues/15

**Comment by DamienCassou** [comment]
```
$ gplaycli --config ~/Documents/projects/gplaycli/credentials.conf --folder ~/Documents/projects/playstore/ --download com.capitainetrain.android --progress
Error while downloading com.capitainetrain.android : 'WrappedSocket' object has no attribute 'close'

$ cat ~/Documents/projects/gplaycli/credentials.conf 
[Credentials]
gmail_password=jesuischarlie1
android_ID=3d716411bf8bc802
gmail_address=fdroidserver@gmail.com
language=fr_FR
```

**Comment by matlink** [comment]
Which distro are you using ?

**Comment by DamienCassou** [comment]
@matlink I'm on nixos. The gplaycli dependencies I have are:

# Python dependencies

- python2.7-gplaycli-0.1.2
- python2.7-pyasn1-0.1.8
- python-2.7.10
- python2.7-setuptools-18.2
- python2.7-clint-0.5.1
- python2.7-Pillow-2.9.0
- python2.7-blessings-1.5.1
- python2.7-args-0.1.0
- python2.7-ndg-httpsclient-0.4.0
- python2.7-pyopenssl-0.15.1
- python2.7-cryptography-1.1.1
- python2.7-six-1.10.0
- python2.7-idna-2.0
- python2.7-ipaddress-1.0.15
- python2.7-cffi-1.3.0
- python2.7-pycparser-2.14
- python2.7-pyasn1-modules-0.0.5
- python-sqlite3-2.7.10
- python2.7-enum34-1.0.4
- python2.7-protobuf-2.6.1
- python2.7-google-apputils-0.4.1
- python2.7-pytz-2015.4
- python2.7-gflags-2.0
- python2.7-dateutil-2.4.2
- python2.7-mox-0.5.3
- python2.7-requests-1.2.3


# Binary dependencies
- python2.7-gplaycli-0.1.2
- python-2.7.10
- python2.7-setuptools-18.2
- python2.7-Pillow-2.9.0
- python2.7-ndg-httpsclient-0.4.0
- protobuf-2.6.1
- python2.7-gflags-2.0
- python2.7-gplaycli-0.1.2

**Comment by matlink** [comment]
Did you used `pip` to install `gplaycli` ?

**Comment by raphael0202** [comment]
The bug comes from the googleplay.py external dependency, I guess it would be a good thing to print the stack trace when an exception is raised instead of just naming the exception, so we could know where it comes from.

And setting up a logger would be definitely great, as we could have more info on the process leading to the bug.

**Comment by DamienCassou** [comment]
@matlink nix used the setup.sh to check the dependencies. But I can't just use pip like that on nixos because nix is a functional package manager.
@raphael0202 that's what I did in issue #15. My 2 issues got mixed up. Sorry about that. If you can help me there, that would be great. What do you mean by a logger and how can I setup one?

**Comment by matlink** [comment]
Well I think the libraries you use are not exactly the same as I do on a dpkg-distro.

By logger he meant I should include a way to easily get the traceback when an error occurs. That is not your matter.

**Comment by raphael0202** [comment]
After a bit of research it seems that the issue has to do with the SSL connection, more specifically the SSL library requests uses internally to establish the secured connection.

**Comment by DamienCassou** [comment]
I managed to make gplaycli download an apk using a mix of pip and nixos. I used pip for everything but the following dependencies for which I used nix. I did that because gplaycli depends indirectly on `cryptography` which I can't install easily in my pip environment under nixos.

Nix dependencies:

- python2.7-pycparser-2.14
- python2.7-ipaddress-1.0.15
- python2.7-enum34-1.0.4
- python2.7-cffi-1.3.0
- python2.7-cryptography-1.1.1
- python2.7-idna-2.0

I really need to get rid of all pip dependencies and move them towards nix dependencies. I will try to understand which pip package is different from the one in nix.

**Comment by DamienCassou** [comment]
I've just replaced `requests` by `requests2` and things seem to work fine (with default credentials only). Thank you very much

**Comment by DamienCassou** [comment]
I will propose that as a pull request:

```diff
diff --git a/setup.py b/setup.py
index 1f9567f..afc922e 100644
--- a/setup.py
+++ b/setup.py
@@ -22,7 +22,7 @@ setup(name='GPlayCli',
             ['/etc/gplaycli/', ['credentials.conf','cron/cronjob']],
         ],
         install_requires=[
-                'requests',
+                'requests >= 2.0.0',
                 'protobuf',
                 'ndg-httpsclient',
                 'clint',
```

**Comment by matlink** [comment]
Yes  `request` is not the library used, `request2` is. That is why I prefer to install `gplaycli` via `pip`

**Comment by DamienCassou** [comment]
After merging pull request #20, I propose that we close issues #15 and #16.

**Comment by matlink** [comment]
Solved by #20 .
